common/djangoapps/student/models.py

```diff
@@ -1381,6 +1384,22 @@ def refundable(self):
         else:
             return True
 
+    def _refund_window_end_date(self):
+        """ Calculate and return the refund window end date. """
+        order_number = CourseEnrollmentAttribute.get_enrollment_attribute(
+            self,
+            namespace='order',
+            name='order_number'
+        ).get('value')
+
+        order = ecommerce_api_client(self.user).orders(order_number).get()
+
+        refund_window_start_date = max(
+            datetime.strptime(order.get('date_placed', "1970-01-01T00:00:00Z"), "%Y-%m-%dT%H:%M:%SZ"),
```

Comment:
If the date_placed value is missing, there is an even bigger issue with our API or data. Don't provide a default. Definitely let the exception be raised.